integration-tests/smoke/vrfv2plus_test.go

```diff
@@ -1364,6 +1367,211 @@ func TestVRFV2PlusWithBHS(t *testing.T) {
+	linkToken, err := actions.DeployLINKToken(env.ContractDeployer)
+	require.NoError(t, err, "error deploying LINK contract")
+
+	// 1. Add job spec with requestTimeout = 5 seconds
+	timeout := time.Duration(time.Second * 5)
+	numberOfTxKeysToCreate := 2
+	config.VRFv2Plus.General.VRFJobRequestTimeout = &blockchain.StrDuration{Duration: timeout}
+	subBalance := float64(0)
+	config.VRFv2Plus.General.SubscriptionFundingAmountLink = &subBalance
+	config.VRFv2Plus.General.SubscriptionFundingAmountNative = &subBalance
+	vrfv2PlusContracts, subIDs, vrfv2PlusData, nodesMap, err := vrfv2plus.SetupVRFV2_5Environment(
+		env,
+		[]vrfcommon.VRFNodeType{vrfcommon.VRF},
+		&config,
+		linkToken,
+		mockETHLinkFeed,
+		numberOfTxKeysToCreate,
+		2,
+		1,
+		l,
+	)
+	require.NoError(t, err, "error setting up VRF v2_5 env")
+
+	subID := subIDs[0]
+
+	subscription, err := vrfv2PlusContracts.CoordinatorV2Plus.GetSubscription(testcontext.Get(t), subID)
+	require.NoError(t, err, "error getting subscription information")
+
+	vrfv2plus.LogSubDetails(l, subscription, subID, vrfv2PlusContracts.CoordinatorV2Plus)
+
+	t.Run("Timed out request fulfilled after node restart with replay", func(t *testing.T) {
+		configCopy := config.MustCopy().(tc.TestConfig)
+		var isNativeBilling = false
+
+		jobRunsBeforeTest, err := nodesMap[vrfcommon.VRF].CLNode.API.MustReadRunsByJob(nodesMap[vrfcommon.VRF].Job.Data.ID)
+		require.NoError(t, err, "error reading job runs")
+
+		// 2. create request but without fulfilment - e.g. simulation failure (insufficient balance in the sub, )
+		vrfv2plus.LogRandRequest(
+			l,
+			vrfv2PlusContracts.VRFV2PlusConsumer[0].Address(),
+			vrfv2PlusContracts.CoordinatorV2Plus.Address(),
+			subID,
+			isNativeBilling,
+			*configCopy.VRFv2Plus.General.MinimumConfirmations,
+			*configCopy.VRFv2Plus.General.CallbackGasLimit,
+			*configCopy.VRFv2Plus.General.NumberOfWords,
+			vrfv2PlusData.KeyHash,
+			*configCopy.VRFv2Plus.General.RandomnessRequestCountPerRequest,
+			*configCopy.VRFv2Plus.General.RandomnessRequestCountPerRequestDeviation,
+		)
+		_, err = vrfv2PlusContracts.VRFV2PlusConsumer[0].RequestRandomness(
+			vrfv2PlusData.KeyHash,
+			subID,
+			*configCopy.VRFv2Plus.General.MinimumConfirmations,
+			*configCopy.VRFv2Plus.General.CallbackGasLimit,
+			isNativeBilling,
+			*configCopy.VRFv2Plus.General.NumberOfWords,
+			*configCopy.VRFv2Plus.General.RandomnessRequestCountPerRequest,
+		)
+		require.NoError(t, err, "error requesting randomness")
+		initialReqRandomWordsRequestedEvent, err := vrfv2PlusContracts.CoordinatorV2Plus.WaitForRandomWordsRequestedEvent(
+			[][32]byte{vrfv2PlusData.KeyHash},
+			[]*big.Int{subID},
+			[]common.Address{common.HexToAddress(vrfv2PlusContracts.VRFV2PlusConsumer[0].Address())},
+			time.Minute*1,
+		)
+		require.NoError(t, err, "error waiting for initial request RandomWordsRequestedEvent")
+
+		// 3. create new request in a subscription with balance and wait for fulfilment
+		fundingAmt := big.NewFloat(100)
```

Comment:
could you please leave a todo here - we need this to be parametrized, since these tests will be run on live testnets as well. 
just for precaution, lower the value to 5 Link and 0.1 ETH - we should use separate variables for native and link there

Reply:
Done